src/pykeen/nn/representation.py

```diff
@@ -985,3 +986,84 @@ def _plain_forward(
             labels=[self.labels[i] for i in uniq.tolist()],
         )
         return x[inverse]
+
+
+class CombinedRepresentation(Representation):
+    """A combined representation."""
+
+    # TODO: upgrade: NodePiece; Literal
+
+    #: the base representations
+    base: Sequence[Representation]
+
+    #: the combination module
+    # TODO: currently, this is a callable
+    combination: nn.Module
+
+    def __init__(
+        self,
+        max_id: int,
+        base: OneOrManyHintOrType[Representation] = None,
+        base_kwargs: OneOrManyOptionalKwargs = None,
+        combination: Hint[Callable[[Sequence[torch.FloatTensor]], torch.FloatTensor]] = None,
```

Comment:
do we have a combination resolver?

Reply:
e.g., maybe include a simple learnable combination like MLP or some kind of convolution-based combinations

Reply:
for now, I used https://github.com/pykeen/pykeen/pull/964/commits/cebb102530214340396a36e87a072178260932c7 - this may be sub-optimal, since we do not have support for e.g., the reduction necessary in NodePiece where we have two *sequences* of potentially different length of same-sized vectors